### PR TITLE
fix: fix curl import error when curly braces are present as part of URL

### LINKED ISSUE #19846: [Bug]: cUrl import fails when url contains curly brackets around variable names
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

When creating a new API from cUrl import, the import fails if there are curly brackets in the url. 

Many REST APIs have a record ID in the endpoint path, and the docs will use variables when providing cUrl examples to copy. 

Example:
`/rest/v1/product/{product_id}/variants`

### Steps To Reproduce

The Baserow API docs provide the following cUrl which fails to import unless the brackets are removed. 
```
curl \
-X GET \
-H "Authorization: Token YOUR_DATABASE_TOKEN" \
[https://api.baserow.io/api/database/rows/table/130952/{row_id}/?user_field_names=true](https://api.baserow.io/api/database/rows/table/130952/%7Brow_id%7D/?user_field_names=true)
```

### Public Sample App

_No response_

### Issue video log

_No response_

### Version

Cloud v1.9

### PR DESCRIPTION
**Pull request by sumitsum**
## Description
- so far we have been using the `java.net.URI` object to extract query params from a URL string. However, the creation of URI object throws error if the URL string has curly braces in it. Hence, this PR modifies the process of query params extraction such that it does not need to create a URI object anymore. 
- credits: some parts of the change in this PR is taken from https://github.com/appsmithorg/appsmith/pull/23680 . Thanks to Piyush for raising the earlier PR. 

#### PR fixes following issue(s)
Fixes #19846 

#### Type of change
- Bug fix (non-breaking change which fixes an issue)

#### How Has This Been Tested?
- [x] Manual
- [x] Junit
>
>
#### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

### CODE REVIEW

File: app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/CurlImporterServiceCEImpl.java [hunk 1]
```diff
@@ -521,6 +510,27 @@ private void trySaveURL(ActionDTO action, String token) throws MalformedURLExcep
...
+
+        /**
+         * Attempt to extract query params only if the query params string is non-empty, and it has at least one key
+         * value pair.
+         */
+        if (!isBlank(queryParamsString) && queryParamsString.contains("=")) {
+            Arrays.stream(queryParamsString.split("&"))
+                    .forEach(queryParam -> {
+                        String[] paramMap = queryParam.split("=", 2);
+                        if (paramMap.length > 1) {
```
Comment (sondermanish) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/CurlImporterServiceCEImpl.java:525]: these strings could be made constants, for multiple usage|
coudl be tackled in a separate PR?

**Review by sondermanish — DISMISSED** [review]
LGTM

File: app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/CurlImporterServiceCEImpl.java (hunk 1 above)
Comment (sumitsum) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/CurlImporterServiceCEImpl.java:525]: Thanks for the suggestion @sondermanish . There are multiple usage of magic strings across the file. We can take care of those as well. Creating an issue to track it: https://github.com/appsmithorg/appsmith/issues/25581

**Comment by prapullc** [comment]
All the scenario wrt above is working as expected.
@sumitsum 
 However when we try to add `curl http://site.{one, two, three}.com` there should be an error message displayed which is not handled and improper API gets converted

**Comment by sumitsum** [comment]
@prapullc I am seeing the url being rejected as shown in the video: 

https://github.com/appsmithorg/appsmith/assets/1757421/e4091722-ba4c-41a9-b058-c63d95592dd7

**Comment by sumitsum** [comment]
> All the scenario wrt above is working as expected. @sumitsum However when we try to add `curl http://site.{one, two, three}.com` there should be an error message displayed which is not handled and improper API gets converted

As discussed with Prapulla, the issue seems to be already present on prod as well and does not look related to this PR, hence moving forward to merge this PR.

**Pull request merged**